docs/rfcs/042-compute-pageserver-communicator.md

```diff
@@ -0,0 +1,197 @@
+# Compute <-> Pageserver Communicator Rewrite
+
+## Summary
+
+## Motivation
+
+- prefetching logic is complicated
+- handling async libpq connections in C code are difficult and error-prone
+- only few people are comfortable working on the code
+- new AIO (maybe) coming up in Postgres v18
+- cannot process prefetch replies until another I/O function is called. Makes it impossible to accurately measure when a reply was received
+- every backend opens a separate connection to pageservers -> lots of connections, first query in backend is slow
+
+- desire for better protocol, not libpq-based
+
+- By writing the "communicator" as a separate rust module, it can be reused in
+  tests, outside PostgreSQL.
+
+## Non Goals (if relevant)
+
+- We will keep LFC unmodified for now. It might be a good idea to rewrite it
```

Comment:
If we have separate communication process (through which all `get page` requests will be passed), then it seems to be quite easy to implement caching (in memory on on disk) in this process. It can be done in local memory and so easily scaled up and down. Also, if requests will be fetched from request buffers by single thread, then we do not need synchronisation (but I am not sure if it is good idea).

Reply:
Yeah, we could move the LFC into the communicator process. It would add some delay to LFC cache hits though, as you'd need to go through the shared memory request/response even for cache hits. It might still be worth it, if it makes the LFC management simpler or more flexible.

Anyway, I don't want to tackle that right now.